rllib/evaluation/sampler.py

```diff
@@ -967,10 +967,12 @@ def _process_observations(
             # Horizon hit and we have a soft horizon (no hard env reset).
             if hit_horizon and soft_horizon:
                 episode.soft_reset()
-                resetted_obs: Dict[AgentID, EnvObsType] = all_agents_obs
+                resetted_obs: Dict[EnvID, Dict[AgentID, EnvObsType]] = \
```

Comment:
This is a little hacky, but I didn't want to modify the sampler code too much here.